lib/tapioca/dsl/compilers/active_model_validations_confirmation.rb

```diff
@@ -0,0 +1,97 @@
+# typed: strict
+# frozen_string_literal: true
+
+begin
+  require "active_model"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Dsl
+    module Compilers
+      # `Tapioca::Dsl::Compilers::ActiveModelValidationsConfirmation` decorates RBI files for all
+      # classes that use [`ActiveModel::Validates::Confirmation`](https://api.rubyonrails.org/classes/ActiveModel/Validations/HelperMethods.html#method-i-validates_confirmation_of).
+      #
+      # For example, with the following class:
+      #
+      # ~~~rb
+      # class User
+      #   include ActiveModel::Validations
+      #
+      #   validates_confirmation_of :password
+      #
+      #   validates :email, confirmation: true
+      # end
+      # ~~~
+      #
+      # this compiler will produce an RBI file with the following content:
+      # ~~~rbi
+      # # typed: true
+      #
+      # class User
+      #
+      #   sig { returns(T.untyped) }
+      #   def email_confirmation; end
+      #
+      #   sig { params(email_confirmation=: T.untyped).returns(T.untyped) }
+      #   def email_confirmation=(email_confirmation); end
+      #
+      #   sig { returns(T.untyped) }
+      #   def password_confirmation; end
+      #
+      #   sig { params(password_confirmation=: T.untyped).returns(T.untyped) }
+      #   def password_confirmation=(password_confirmation); end
+      # end
+      # ~~~
+      class ActiveModelValidationsConfirmation < Compiler
+        extend T::Sig
+
+        ConstantType = type_member do
+          {
+            fixed: T.all(
+              T::Class[ActiveModel::Validations],
+              ActiveModel::Validations::HelperMethods,
+              ActiveModel::Validations::ClassMethods,
+            ),
+          }
+        end
+
+        class << self
+          sig { override.returns(T::Enumerable[Module]) }
+          def gather_constants
+            # Collect all the classes that include ActiveModel::Validations
+            all_classes.select do |c|
+              c < ActiveModel::Validations &&
+                c.name.to_s != "ActiveRecord::Base"
+            end
+          end
+        end
+
+        sig { override.void }
+        def decorate
+          confirmation_validators = constant.validators.select do |v|
+            v.is_a?(ActiveModel::Validations::ConfirmationValidator)
+          end
```

Comment:
I _think_ you can do this.
```suggestion
          confirmation_validators = constant.validators.grep(ActiveModel::Validations::ConfirmationValidator)
```